portal-impl/src/com/liferay/portal/servlet/filters/language/LanguageFilter.java

```diff
@@ -118,13 +97,20 @@ protected void processFilter(
 
 		content = translateResponse(httpServletRequest, content);
 
-		eTagValue =
+		String eTag =
 			StringPool.QUOTE + DigesterUtil.digest("SHA-1", content) +
 				StringPool.QUOTE;
 
-		_eTagValues.put(eTagKey, eTagValue);
+		httpServletResponse.setHeader(HttpHeaders.ETAG, eTag);
+
+		String ifNoneMatch = httpServletRequest.getHeader(
+			HttpHeaders.IF_NONE_MATCH);
 
-		httpServletResponse.setHeader(HttpHeaders.ETAG, eTagValue);
+		if ((ifNoneMatch != null) && ifNoneMatch.equals(eTag)) {
+			httpServletResponse.setStatus(HttpServletResponse.SC_NOT_MODIFIED);
+
+			return;
+		}
```

Comment:
What's the use case to do this after processFilter? Is this covering the case where translations change at runtime  right?

Reply:
This change will impact the amount we send over the network. We have to compute the etag everytime, so the server still runs this computation, but if the etag doesn't change then we just tell the client that the data isn't modfied and will have faster network speeds.

Ideally we can update this down the road so that the server doesn't have to translate every time, that is the next step